waku/waku_sync/raw_bindings.nim

```diff
@@ -6,173 +6,234 @@ else:
+proc raw_initiate(negentropy: pointer, output: ptr Buffer): int  {.header: NEGENTROPY_HEADER, importc: "negentropy_initiate".}
 
 # https://github.com/hoytech/negentropy/blob/6e1e6083b985adcdce616b6bb57b6ce2d1a48ec1/cpp/negentropy.h#L58
-proc raw_setInitiator(this: Negentropy) {.importcpp: "negentropy::setInitiator".}
+proc raw_setInitiator(negentropy: pointer) {.header: NEGENTROPY_HEADER, importc: "negentropy_setinitiator".}
 
 # https://github.com/hoytech/negentropy/blob/6e1e6083b985adcdce616b6bb57b6ce2d1a48ec1/cpp/negentropy.h#L62
-proc raw_reconcile(this: Negentropy, query: String): String {.importcpp: "negentropy::reconcile".}
+proc raw_reconcile(negentropy: pointer, query: ptr Buffer, output: ptr Buffer): int {.header: NEGENTROPY_HEADER, importc: "reconcile".}
+#[ 
+type
+  ReconcileCallback = proc(have_ids: ptr Buffer, have_ids_len:uint64, need_ids: ptr Buffer, need_ids_len:uint64, output: ptr Buffer, outptr: var ptr cchar) {.cdecl, raises: [], gcsafe.}# {.header: NEGENTROPY_HEADER, importc: "reconcile_cbk".}
+ ]#
 
 # https://github.com/hoytech/negentropy/blob/6e1e6083b985adcdce616b6bb57b6ce2d1a48ec1/cpp/negentropy.h#L69
-proc raw_reconcile(this: Negentropy, query: String, haveIds: var Vector[String], needIds: var Vector[String]): String {.importcpp: "negentropy::reconcile".}
+#proc raw_reconcile(negentropy: pointer, query: ptr Buffer, cbk: ReconcileCallback, output: ptr cchar): int {.header: NEGENTROPY_HEADER, importc: "reconcile_with_ids".}
+
+proc raw_reconcile(negentropy: pointer, query: ptr Buffer, r: ptr BindingResult){.header: NEGENTROPY_HEADER, importc: "reconcile_with_ids_no_cbk".}
+
+proc free_result(r: ptr BindingResult){.header: NEGENTROPY_HEADER, importc: "free_result".}
 
 ### Wrappings ###
 
-proc new*(T: type Storage): T =
-  let storage = Storage()
+#TODO: Change all these methods to private as we don't want them to be exposed outside Sync package
+#TODO: Wrap storage and negentropy with objects rather than using void pointers
+proc new_storage*(): pointer =
+  let storage = storage_init("", "")
 
   return storage
 
-proc erase*(self: Storage, id: int64, hash: WakuMessageHash) =
-  let cppString = String.fromWakuMessageHash(hash)
+proc erase*(storage: pointer, id: int64, hash: WakuMessageHash): bool =
+  let cString = toBuffer(hash)
   
-  self.raw_erase(clong(id), cppString)
+  return raw_erase(storage, uint64(id), cString.unsafeAddr)
 
-proc insert*(self: Storage, id: int64, hash: WakuMessageHash) =
-  let cppString = String.fromWakuMessageHash(hash)
-  
-  self.raw_insert(clong(id), cppString)
+proc insert*(storage: pointer, id: int64, hash: WakuMessageHash): bool =
+  var buffer = toBuffer(hash)
+  var bufPtr = addr(buffer)
+  return raw_insert(storage, uint64(id), bufPtr)
 
-proc new*(T: type Negentropy, storage: Storage, frameSizeLimit: uint64): T =
-  let negentropy = constructNegentropy(storage, culong(frameSizeLimit))
+proc new_negentropy*(storage: pointer, frameSizeLimit: uint64): pointer =
```

Comment:
Suggestion that encourages to follow similar naming convention as mentioned above by just prepending the `negentropy` word. In this case:
```suggestion
proc negentropyNew*(storage: pointer, frameSizeLimit: uint64): pointer =
```